test/configuration_test.dart

```diff
@@ -130,12 +132,105 @@ msix_config:
     test('fallback to pubspec version', () async {
       await File(yamlTestPath).writeAsString(
         'name: testAppWithVersion\n'
-        'version: 1.1.3',
+        'version: 1.2.3',
       );
       await config.getConfigValues();
-      expect(config.msixVersion, equals('1.1.3.0'));
+      expect(config.msixVersion, equals('1.2.3.0'));
+    });
+    test('pubspec with buildnumber', () async {
+      await File(yamlTestPath).writeAsString(
+        'name: testAppWithVersion\n'
+        'version: 1.2.3+45',
+      );
+      await config.getConfigValues();
+      expect(config.msixVersion, equals('1.2.3.0'));
+    });
+    group("versionWithBuildNumber:", () {
+      const _testList = <VersionTestCase>[
+        VersionTestCase(
+          testName: "normal version",
+          versionString: "1.2.3+4",
+          versionWithBuildNumber: false,
+          result: "1.2.3.0",
+        ),
+        VersionTestCase(
+          testName: "normal version",
+          versionString: "1.2.3+4",
+          versionWithBuildNumber: true,
+          result: "1.203.4.0",
+        ),
+        VersionTestCase(
+          testName: "maximum major version",
+          versionString: "65535.2.3+4",
+          versionWithBuildNumber: true,
+          result: "65535.203.4.0",
+        ),
+        VersionTestCase(
+          testName: "major version too high",
+          versionString: "65536.2.3+4",
+          versionWithBuildNumber: true,
+          result: null,
+        ),
+        VersionTestCase(
```

Comment:
Should probably test with maximum patch as well (or new test with `5.654.99+0`)

Reply:
I have added a new test:
https://github.com/YehudaKremer/msix/blob/c2059a49b6feb5606df10147f59a68ac6ea2e95e/test/configuration_test.dart#L228-L233